lib/cosmos/interfaces/protocols/crc_protocol.rb

```diff
@@ -0,0 +1,133 @@
+# encoding: ascii-8bit
+
+# Copyright 2017 Ball Aerospace & Technologies Corp.
+# All Rights Reserved.
+#
+# This program is free software; you can modify and/or redistribute it
+# under the terms of the GNU General Public License
+# as published by the Free Software Foundation; version 3 with
+# attribution addendums as found in the LICENSE.txt
+
+require 'cosmos/config/config_parser'
+require 'cosmos/interfaces/protocols/protocol'
+require 'cosmos/utilities/crc'
+require 'thread'
+
+module Cosmos
+  # Creates a CRC on write and verifies a CRC on read
+  class CrcProtocol < Protocol
+    ERROR = "ERROR" # on CRC mismatch
+    DISCONNECT = "DISCONNECT" # on CRC mismatch
+
+    def initialize(write_item_name, strip_crc, bad_strategy, bit_offset,
+                   bit_size = 32, poly = nil, seed = nil, xor = nil, reflect = nil)
+      @write_item_name = ConfigParser.handle_nil(write_item_name)
+      @strip_crc = ConfigParser.handle_true_false(strip_crc)
+      raise "Invalid strip CRC of '#{strip_crc}'. Must be TRUE or FALSE." unless !!@strip_crc == @strip_crc
```

Comment:
handle_true_false returns the value as is if it's not true or false. I'm checking that it's actually true or false.